core/shared/src/main/scala/scalaz/zio/ZIO.scala

```diff
@@ -1593,6 +1593,24 @@ trait ZIOFunctions extends Serializable {
    * current fiber for this operation to return non-empty lists.
    */
   final def children: UIO[IndexedSeq[Fiber[_, _]]] = descriptor.flatMap(_.children)
+
+  def fromOptionM[A](uio: UIO[Option[A]]): IO[Unit, A] = uio.flatMap {
```

Comment:
This looks great and the implementation is 100% correct!

Note there is a slightly simpler way to write this:

```scala
uio.flatMap(fromOption(_))
```

The `fromOptionM` function generates an `IO[Unit, A]`, which does not have the same signature as `UIO` (the `IO` can fail, but all `UIO` methods cannot fail). 

For this reason, we should move this method to the `ZIO_E_Any` trait, which also has `fromOption`.

Reply:
What if you want to convert a `Task[Option[A]]` to a `IO[E, A]` ? This is quite common as a query to the database usually returns a `Task`. I think a function like this would also be quite useful:

```scala
def fromOptionTask[A](task: Task[Option[A]], ifEmpty: => E): IO[E, A]
```

Reply:
Since there's already `get` that does the same thing as `fromOptionM`, how about adding `getOrElse`? It could be used both for giving a fallback value or an error.